community/kernel/src/main/java/org/neo4j/kernel/configuration/BoltConnectorValidator.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.neo4j.kernel.configuration;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.Nonnull;
+
+import org.neo4j.graphdb.config.InvalidSettingException;
+import org.neo4j.graphdb.config.Setting;
+import org.neo4j.helpers.ListenSocketAddress;
+
+import static org.neo4j.kernel.configuration.BoltConnector.EncryptionLevel.OPTIONAL;
+import static org.neo4j.kernel.configuration.Connector.ConnectorType.BOLT;
+import static org.neo4j.kernel.configuration.Settings.BOOLEAN;
+import static org.neo4j.kernel.configuration.Settings.NO_DEFAULT;
+import static org.neo4j.kernel.configuration.Settings.advertisedAddress;
+import static org.neo4j.kernel.configuration.Settings.listenAddress;
+import static org.neo4j.kernel.configuration.Settings.options;
+import static org.neo4j.kernel.configuration.Settings.setting;
+
+public class BoltConnectorValidator extends ConnectorValidator
+{
+    public BoltConnectorValidator()
+    {
+        super( BOLT );
+    }
+
+    @Override
+    @Nonnull
+    public Map<String,String> validate( @Nonnull Map<String,String> rawConfig )
+            throws InvalidSettingException
+    {
+        final HashMap<String,String> result = new HashMap<>();
+
+        ownedEntries( rawConfig ).forEach( s ->
```

Comment:
Not without adding a great deal of obfuscation in my opinion, which I consider much worse